src/Points.ts

```diff
@@ -0,0 +1,146 @@
+interface ITotalOfChar {
+  letters: {
+    uppercase: number;
+    lowercase: number;
+  };
+  numbers: number;
+  symbols: number;
+}
+
+type THeightPoints = 25 | 10 | 5 | 0;
+type TMediumPoints = 20 | 10 | 5 | 0;
+type TLowPoints = 5 | 2 | 3 | 0;
+
+interface IPoints {
+  length: () => THeightPoints;
+  letters: () => TMediumPoints;
+  numbers: () => TMediumPoints;
+  characters: () => THeightPoints;
+  bonus: () => TLowPoints;
+  getAll: () => number;
+}
+
+class Points implements IPoints {
+  private passwordLength: number;
+
+  private totalOfChar: ITotalOfChar;
+
+  constructor(password: string, validSymbolsChars?: string) {
+    const symbolsChar =
+      validSymbolsChars || password.split(/[A-z]|[0-9]/g).join('');
+
+    const symbolsSearchRegex: RegExp = new RegExp(
+      `\\${symbolsChar.split('').join('|\\')}`, // used \\ to treat special regex characters (*!$+) as search instead of validation
+      'g'
+    );
+
+    this.passwordLength = password.length;
+
+    this.totalOfChar = {
+      letters: {
+        lowercase: password.split(/[a-z]/g).length,
+        uppercase: password.split(/[A-Z]/g).length,
+      },
+      numbers: password.split(/[0-9]/g).length,
+      symbols: password.split(symbolsSearchRegex).length,
+    };
+  }
+
+  length() {
+    switch (true) {
```

Comment:
I think that is more coll use an if, and else if, else sequence because the switch is used to "options"

Reply:
I prefer to change because it’s leaner, without {}

Reply:
Ok...